### PR TITLE
Bump required QEMU version to 6.2.0 and build CI on Ubuntu 22.04

### PR DESCRIPTION
**Pull request by nathanbw**
This PR updates BuildInstructions.md to bump the minimum required QEMU version to 6.2

It also updates Meta/run.sh to check for the new QEMU version and add support for minor release numbers in the version checking logic

The change to Meta/run.sh caused CI to fail, since the Github Actions workers were using the default version of QEMU in ubuntu-20.04, which was 6.0, not 6.2. In order to get CI to pass, I went ahead and updated the Github Actions workers to ubuntu-22.04

Once I updated the Github Actions workers to ubuntu-22.04, the build started failing again due to a few lints, so I made little fixes for those

### CODE REVIEW

File: Documentation/BuildInstructions.md [hunk 1]
```diff
@@ -31,10 +31,12 @@ Now on Ubuntu or Debian you can install gcc-11 with apt like this:
 sudo apt install gcc-11 g++-11
 ```
 
-#### QEMU 5 or later
+#### QEMU 6.2.0 or later
 
-QEMU version 5 is available in Ubuntu 20.10, but it is recommended to build Qemu as provided by the toolchain by running `Toolchain/BuildQemu.sh`.
-Note that you might need additional dev packages:
+Version 6.0 of Qemu is available as a package in Ubuntu 21.10, but version 6.2.0 or later is required to run
```
Comment (timschumi) [inline Documentation/BuildInstructions.md:36]: I'd just leave out the part about Ubuntu 21.10, I don't think there is a point in specifically mentioning that an outdated Ubuntu version provides an outdated QEMU verison. If you want to mention which QEMU version is provided by Ubuntu, do so for the newest LTS (i.e. 22.04), which provides QEMU 6.2.

The remaining text can stay as-is in my opinion, except for maybe replacing `Qemu` with `QEMU`.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Documentation/BuildInstructions.md (hunk 1 above)
Comment (nathanbw) [reply Documentation/BuildInstructions.md]: Thanks for your review -- I have updated the PR. I force-pushed the branch, which I realize may have been a faux pas. (I saw that the commit-lint CI step had failed, probably due to a longer-than-72-chars commit message, and I wasn't sure whether the PR would be merge-able with that lint failure -- new contribution jitters)

Comment (timschumi) [reply Documentation/BuildInstructions.md]: Force-pushes are absolutely fine. :^)

**Comment by timschumi** [comment]
I just noticed that the `run.sh` script still checks for QEMU 5. That might also be something that we should be updating at the same time.

https://github.com/SerenityOS/serenity/blob/master/Meta/run.sh#L106-L117

**Comment by nathanbw** [comment]
@timschumi I added https://github.com/SerenityOS/serenity/pull/13776/commits/7f2c992cc715303805b26a6836474998e20d7ded to check for QEMU 6.2 or later. I tested the changes in `run.sh` by copying those stanzas into another script and then setting `SERENITY_QEMU_BIN` to a script that would output a fake QEMU version number. I tested and got the correct result for QEMU versions 5.6, 6.0.0, 6.2.0, 6.1.9, and 7.0.0, so I think it works :^)

**Comment by nathanbw** [comment]
It seems this PR uncovered a new yak to shave: the build now fails because the test workers have QEMU 6.0.0 🤦‍♂️ (see [here](https://github.com/SerenityOS/serenity/runs/6167825789?check_suite_focus=true) and [here](https://github.com/SerenityOS/serenity/runs/6167825743?check_suite_focus=true))

Also, any opinion on whether I ought to update this PR to check for version 7.0 or above? (See [my comment](https://github.com/SerenityOS/serenity/issues/13771#issuecomment-1109196934) on https://github.com/SerenityOS/serenity/issues/13771)

**Comment by EWouters** [comment]
It seems to me the easiest way to solve this is to upgrade to Ubuntu 22.04. Alternatively we can look into which change caused us not to support QEMU 6.0.0 anymore.

**Comment by nathanbw** [comment]
> It seems to me the easiest way to solve this is to upgrade to Ubuntu 22.04

I hadn't considered that, but it does seems the easiest path forward. Though I think we'll have to wait for https://github.com/actions/virtual-environments/discussions/5427

> Alternatively we can look into which change caused us not to support QEMU 6.0.0 anymore

Here's the commit that broke QEMU 6.0.0 for me (with the commit in place, `/dev/keyboard0` isn't created in the SerentiyOS VM during boot, so typing doesn't work): https://github.com/SerenityOS/serenity/commit/063ea0088ed4798fc3be7dc47ce4e89dbf3e26f3. I'm not sure how vital that commit is (maybe @supercomputer7 can weigh in?)

Thanks all for your reviews!

**Review by BertalanD — APPROVED** [review]

File: Meta/Azure/Setup.yml [hunk 2]
```diff
@@ -7,20 +7,17 @@ steps:
...
 
   - ${{ if eq(parameters.os, 'Linux') }}:
     - script: |
         sudo apt-get purge -y clang-11 clang-12 gcc-10
-        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
         wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-        sudo add-apt-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main'
+        sudo add-apt-repository 'deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-13 main'
         sudo apt-get update
         sudo apt-get install ccache gcc-11 g++-11 clang-13 libstdc++-11-dev ninja-build unzip
```
Comment (trflynn89) [inline Meta/Azure/Setup.yml:22]: (Not related to this PR, just caught my eye)
@BertalanD Should Lagom CI be using LLVM 14 at this point?

Comment (BertalanD) [reply Meta/Azure/Setup.yml:22]: Sure, why not?

Ubuntu 22.04 ships LLVM 14 by default, but only version 10 and 11 are pre-installed on the VMs. I can open a feature request later today.

**Comment by bgianfo** [comment]
@nathanbw it looks like this might have broken the PVS static analysis build? 
https://github.com/SerenityOS/serenity/runs/6583739343?check_suite_focus=true#step:4:91

**Comment by nathanbw** [comment]
@bgianfo I think you are correct -- I have created https://github.com/SerenityOS/serenity/pull/14098 to try to fix the issue. I'm going to click around and see if I can get that particular workflow to run against my branch to ensure my fix actually works

**Comment by nathanbw** [comment]
Thanks to timschumi for pointing out that I missed https://github.com/SerenityOS/serenity/pull/14097 by EWouters which should fix this 👍

**Pull request merged**